docs/migrate-from-cluster.md

```diff
@@ -0,0 +1,226 @@
+
+    ```ShellSession
+    $ pwd
+    <compliantkubernetes-kubespray repo root>/kubespray/inventory/${ENVIRONMENT_NAME}
+
+    export ENVIRONMENT_NAME="name-of-environment"
+
+    terraform import -var-file cluster.tfvars -state ${ENVIRONMENT_NAME}.tfstate -config=../../contrib/terraform/openstack/ module.network.openstack_networking_network_v2.k8s[0] <ID from same resource above>
+    terraform import -var-file cluster.tfvars -state ${ENVIRONMENT_NAME}.tfstate -config=../../contrib/terraform/openstack/ module.network.openstack_networking_router_v2.k8s[0] <ID from same resource above>
+    terraform import -var-file cluster.tfvars -state ${ENVIRONMENT_NAME}.tfstate -config=../../contrib/terraform/openstack/ module.network.openstack_networking_subnet_v2.k8s[0] <ID from same resource above>
+    terraform import -var-file cluster.tfvars -state ${ENVIRONMENT_NAME}.tfstate -config=../../contrib/terraform/openstack/ module.network.openstack_networking_router_interface_v2.k8s[0] <ID from same resource above>
+    ```
+
+    If imported correctly the plan of the new machines should only modify the imported resources.
+    Nothing should be recreated.
+
+1. Prepare old machines
+
+    Before running kubespray you'll need to make a couple of tweaks on the old machines.
+
+    * All machines
+
+      Remove `/etc/docker/daemon.json`.
+
+      **NOTE**: This might break docker if it's restarted before running kubespray.
+
+      ```ShellSession
+      sudo mv /etc/docker/daemon.json /etc/docker/daemon.json.backup
+      ```
+
+    * Masters only
+
+      Create `/var/log/kube-audit`
+
+      ```ShellSession
+      sudo mkdir var/log/kube-audit
+      ```
+
+      Change ETCD version to `quay.io/coreos/etcd:v3.4.13`
+
+      **NOTE**: This will restart etcd and if you only have one master it will stop the api server until it's restarted.
+
+      ```ShellSession
+      sudo vim /etc/kubernetes/manifest/etcd.yaml
+      ```
+
+1. Create inventory for the old cluster
+
+    Add the old machines to the new kubespray config path.
+
+    Remember to add
+
+    ```ini
+    [kube-master:vars]
+    supplementary_addresses_in_ssl_keys = [ "<LB public IP>", "<LB private IP>" ]
+    ```
+
+1. Modify default group_vars to match old setup
+
+    Add the following to the `k8s-cluster` group
+
+    ```yaml
+    etcd_kubeadm_enabled: true
+    kube_cert_dir: "/etc/kubernetes/pki"
+    kube_service_addresses: "10.96.0.0/12"
+    kube_pods_subnet: "192.168.0.0/16"
+
+    audit_log_hostpath: "/var/log/kube-audit"
+    audit_log_path: "/var/log/kube-audit/kube-apiserver.log"
+    audit_policy_file: "/etc/kubernetes/conf/audit-policy.yaml"
+    ```
+
+1. Run kubespray on the old machines
+
+    ```ShellSession
+    export CK8S_CONFIG_PATH=/path/to/new/config
+
+    bin/ck8s-kubespray run-playbook ${ENVIRONMENT_NAME} cluster.yml
+    ```
+
+    Verify that all kubernetes components are running and healthy after this step.
+
+1. Run terraform to set up new machines
```

Comment:
Perhaps add some more instructions here. E.g. some example terraform snippet

Reply:
I intentionally left some things out so that it's clear that you need to be quite comfortable with how to set up a cluster with ck8s-kubespray before trying to do anything like this. And

Reply:
But I guess it could be a bit clearer if I added a small snippet for this as well